### PR TITLE
Port lma kinematics plugin

### PR DESCRIPTION
**Pull request by anasarrak**

### CODE REVIEW

File: moveit_kinematics/CMakeLists.txt [hunk 1]
```diff
@@ -33,13 +28,25 @@ pluginlib_export_plugin_description_file(moveit_core kdl_kinematics_plugin_descr
 ament_export_libraries(${PROJECT_NAME} kdl_kinematics_plugin)
 
 include_directories(${THIS_PACKAGE_INCLUDE_DIRS}
+  ${rclcpp_INCLUDE_DIRS}
+  ${random_numbers_INCLUDE_DIRS}
+  ${moveit_msgs_INCLUDE_DIRS}
+  ${moveit_core_INCLUDE_DIRS}
+)
+
+include_directories(SYSTEM ${EIGEN3_INCLUDE_DIRS}
```
Comment (ahcorde) [inline moveit_kinematics/CMakeLists.txt:37]: SYSTEM ?

File: moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp [hunk 2]
```diff
@@ -341,11 +334,15 @@ bool LMAKinematicsPlugin::getPositionFK(const std::vector<std::string>& link_nam
   {
     if (fk_solver_->JntToCart(jnt_pos_in, p_out) >= 0)
     {
-      poses[i] = tf2::toMsg(p_out);
```
Comment (ahcorde) [inline moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp:344]: Add a TODO here

File: moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp [hunk 3]
```diff
@@ -88,68 +88,66 @@ bool LMAKinematicsPlugin::initialize(const moveit::core::RobotModel& robot_model
...
-  ROS_DEBUG_NAMED("lma", "LMA solver initialized");
+  RCLCPP_DEBUG(node_->get_logger(), "LMA solver initialized");
   return true;
 }
 
-bool LMAKinematicsPlugin::timedOut(const ros::WallTime& start_time, double duration) const
+bool LMAKinematicsPlugin::timedOut(const std::chrono::system_clock::time_point& start_time, double duration) const
 {
-  return ((ros::WallTime::now() - start_time).toSec() >= duration);
+  return ((std::chrono::system_clock::now() - start_time).count() >= duration);
```
Comment (ahcorde) [inline moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp]: which is the units of duration?

**Review by ahcorde — CHANGES_REQUESTED** [review]

File: moveit_kinematics/CMakeLists.txt (hunk 1 above)
Comment (anasarrak) [reply moveit_kinematics/CMakeLists.txt:37]: Same approach as https://github.com/ros-planning/moveit2/blob/e23dc1d04d8fcabb0e561593c5316dda23c6ab5b/moveit_ros/perception/CMakeLists.txt#L85-L90.

_If the SYSTEM option is given, the compiler will be told the directories are meant as system include directories on some platforms._

File: moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp (hunk 2 above)
Comment (anasarrak) [reply moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp:344]: https://github.com/AcutronicRobotics/moveit2/pull/75/commits/6076b1fa9db1089c9275d0f1b8a4a45e3a1ca441

File: moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp (hunk 3 above)
Comment (anasarrak) [reply moveit_kinematics/lma_kinematics_plugin/src/lma_kinematics_plugin.cpp]: https://github.com/AcutronicRobotics/moveit2/pull/75/commits/c2f4ba6a8fee1de985726e4d0d6156ce9704db36

**Review by ahcorde — APPROVED** [review]

**Review by ahcorde — APPROVED** [review]

**Pull request merged**